### PR TITLE
SCA readiness

### PR DESCRIPTION
**Pull request by willock-stripe**
FAO @stripe/dev-relations, _particularly_ @ctrudeau-stripe!

## Motivation
This PR attempts to upgrade the `stripe-billing-typographic` demo to use the new [`SetupIntents`](https://stripe.com/docs/api/setup_intents) and [`PaymentMethods`](https://stripe.com/docs/api/payment_methods) APIs to ensure the application is ready for the Strong Customer Authentication regulation.

## Changes
- Updated some vulnerable or outdated dependencies in `package.json`
- Updated `stripe` NodeJS SDK to support latest API resources
- Updated Billing-specific API calls for tiered unit amounts
- Replaced `Sources` API calls with `SetupIntents` and `PaymentMethods` for SCA-readiness
- Update Stripe.js integration to support 3D Secure with `handleCardSetup`
- Update usage records API calls

## Open questions
- I removed the `DELETE /sources` endpoint from `server/routes/stripe.js`, and didn't replace it, because it didn't appear to be called anywhere. Is that peaceful?
- Can you work with me on a test plan?
- Should we build an path for immediately-billed subscriptions using `PaymentIntents`, or perhaps demonstrate how to raise a one-off invoice?

### CODE REVIEW

File: server/models/Customer.js [hunk 1]
```diff
@@ -166,67 +166,65 @@ class Customer extends Model {
...
+          {
+            default_payment_method: paymentMethodId,
+            collection_method: collectionMethod,
+          }
         );
+
         const updatedBilling = await db(Subscription.table)
           .where('id', subscription.id)
-          .update({billing: paymentMethod});
+          .update({collectionMethod: collectionMethod});
```
Comment (ctrudeau-stripe) [inline server/models/Customer.js]: With this changing billing to collectionMethod, we need to update collectionMethod for invoice settings on line 251 https://github.com/stripe/stripe-billing-typographic/pull/25/files#diff-7f8dc7dd9c244818d5292c60b3dfcfddR251

**Comment by ctrudeau-stripe** [comment]
Can we update the test card to be the 3DS test card?

**Comment by ctrudeau-stripe** [comment]
LGTM, thank you for updating this.

**Comment by willock-stripe** [comment]
Handle the `pending_setup_intent`, don't generate a new one.

**Comment by ctrudeau-stripe** [comment]
Can we also switch the test card to the 3DS test card?

**Comment by willock-stripe** [comment]
@ctrudeau-stripe – this is ready for a final pass, thanks for your comments so far!

**Comment by ctrudeau-stripe** [comment]
LGTM 

Thank you for updating this!

**Pull request merged**